src/audio/kpb.c

```diff
@@ -661,9 +661,10 @@ static int kpb_copy(struct comp_dev *dev)
 		comp_update_buffer_consume(source, copy_bytes);
 
 		break;
+	case KPB_STATE_INIT_DRAINING:
```

Comment:
The problem is during "KPB_STATE_INIT_DRAINING" there should be no internal buffering as this can corrupt read pointer calculation.

Reply:
@mrajwa 
I agree, but after adding spin lock in #2329, that situation will never happen. We want to proper handle situation when new ```kpb_copy()``` will come between ```kpb_init_draining()``` and ```kpb_draining_task()```.